File: be/src/main/java/com/example/Campung/notification/dto/NotificationResponse.java
package com.example.Campung.notification.dto;

import com.example.Campung.entity.Notification;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class NotificationResponse {
    
    private Long notificationId;
    private String type;
    private String title;
    private String message;
    private String data;
    private Boolean isRead;
    private LocalDateTime createdAt;
    
    public static NotificationResponse from(Notification notification) {
        return NotificationResponse.builder()
                .notificationId(notification.getNotificationId())
                .type(notification.getType())
                .title(notification.getTitle())
                .message(notification.getMessage())
                .data(notification.getData())
                .isRead(notification.getIsRead())
                .createdAt(notification.getCreatedAt())
                .build();
    }
}

File: android/campung/app/src/main/java/com/shinhan/campung/data/remote/response/NotificationResponse.kt
package com.shinhan.campung.data.remote.response

import com.google.gson.annotations.SerializedName

data class NotificationResponse(
    @SerializedName("notificationId")
    val notificationId: Long,
    
    @SerializedName("type")
    val type: String, // "normal", "friendRequest", "locationShareRequest"
    
    @SerializedName("title")
    val title: String,
    
    @SerializedName("message")
    val message: String,
    
    @SerializedName("data")
    val data: String? = null, // JSON 형태의 추가 데이터
    
    @SerializedName("isRead")
    val isRead: Boolean = false,
    
    @SerializedName("createdAt")
    val createdAt: String,
    
    @SerializedName("readAt")
    val readAt: String? = null
)